paperqa/agents/search.py

```diff
@@ -325,6 +331,14 @@ async def _add_document() -> None:
                 f" within {lock_acquisition_max_retries} attempts."
             )
             raise
+        
+    async def commit(self) -> None:
+        """Commit all pending changes to the index."""
+        if self._writer:
+            self._writer.commit()
+            self._writer.wait_merging_threads()
+            self._searcher = None
+            self._writer = None
```

Comment:
I'm not sure if I should introduce this method or instead run this code inside the .save_index(). 
Probably the second one is a better idea.